### ISSUE TITLE
Trained on single batch with 1 gpu get better result than multiple batch and multiple gpus

### ISSUE THREAD

**Issue by zhiwenfan** [noahcao/Pixel2Mesh#13]
Hi, I have used this code to train my own dataset and  it converged.  However trained on single batch with only 1 gpu get better result than multiple batch and multiple gpus.
I have only changed the  dataloader, can you help with this problem? thanks.

**Comment by ultmaster** [comment]
Hi. Thanks for raising this issue.

It's possible that using a small batch size is better on this task. Due to time limit, we haven't tried this one. Could you kindly share your benchmark results when training with only one GPU? And what is your batch size?

**Comment by zhiwenfan** [comment]
> Hi. Thanks for raising this issue.
> 
> It's possible that using a small batch size is better on this task. Due to time limit, we haven't tried this one. Could you kindly share your benchmark results when training with only one GPU? And what is your batch size?

Thank for your reply.
Even I use the default setting of your master branch with train_tf as training list, I use 4 GPU and 4 batches/32 batches.  The 32 batch produce worse result than 4 batches. 

The 32 batch at Epoch 001, Step 000200/068410, the loss is Loss 0.024578599 (0.055198513)
The 4 batch   Epoch 001, Step 001600/068410, the loss is  Loss 0.012411190 (0.025377692)
which should be similar.

**Comment by zhiwenfan** [comment]
could you please tell how many gpus and how many batch do you use to train?

**Comment by ultmaster** [comment]
Our setting is shared in `default/resnet.yml`. We use 8 GPUs, batch size is set to be 64 (8 on each GPU).

As I said, using smaller batch size could be better on this dataset, as smaller batches are easier to break the local minimum and explore more.

**Comment by zhiwenfan** [comment]
> Our setting is shared in `default/resnet.yml`. We use 8 GPUs, batch size is set to be 64 (8 on each GPU).
> 
> As I said, using smaller batch size could be better on this dataset, as smaller batches are easier to break the local minimum and explore more.

Thank for your kind help.
I use the default options.py which indicates options.dataset.shapenet.num_points = 3000
Is is possible since I only use 3000 points for all batch that lead to a bad results?

**Comment by ultmaster** [comment]
Sorry for slow response.

Yes. This is possible. Using more points is likely to improve the performance. We have done similar experiments (not rigorously) indicating similar phenomenon.

**Comment by ultmaster** [comment]
Closing if it's no longer a problem.

**Comment by zshyang** [comment]
Hi,
Do you know how to download the dataset and unzip it from the link below?
https://drive.google.com/open?id=131dH36qXCabym1JjSmEpSQZg4dmZVQid

**Comment by zshyang** [comment]
Hi,
Do you know how to use their provided YAML files and pretrained weights downloaded from here to do the evaluation?
To my understanding, I should use the downloaded meta folder. Then use the provide pretrained weights.
The command I use is 
```
python entrypoint_eval.py --name xxx --options experiments/default/tensorflow.yml --checkpoint datasets/data/pretrained/vgg16-p2m.pth 
```
But it seems that it only generated some meaningless meshes. And the score is really low.
I am not sure what I did wrong.
![Screenshot from 2020-09-22 02-08-41](https://user-images.githubusercontent.com/57812785/93863819-de0eac00-fc78-11ea-91b5-0235f5b4487b.png)

**Comment by gokceay** [comment]
Hi, 

I have trained pixel2mesh with vgg16 using default/tensorflow.yaml for 2 epoch and the results were really bad (on ShapenetP2M dataset). I have used 8 GPU, batch size of 8 for training and testing. Number of points: 9000 . After reading this post I put a simulation of 100 epoch. How many epoch should I run to obtain the results in the paper? 

Thanks in advance
